src/_utils/mediaSizeProvider/MediaSizeProvider.unit.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import MediaSizeProvider, { MediaSizeContext } from './MediaSizeProvider'
+
+const mockWindowInnerWidth = value => {
+  Object.defineProperty(window, 'innerWidth', {
```

Comment:
nit: did you try with `jest.spyOn(window, 'innerWidth').mockReturnValue({ ... })`

Then it's easy to reset

```
  let spy
  beforeEach(() => {
    spy = jest.spyOn(...
  })

  afterEach(() => {
    if (spy) {
      spy.mockReset()
      spy.mockRestore()
    }
  })
```

Reply:
Ah that was nice I tried it, but spyOn only accept a method as the second parameter, here `innerWidth` is only a readonly property of `window`